src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -136,12 +136,12 @@ protected void FinishDocumentSetup(Document doc, ref RetrieverInput retrieverInp
 
         protected abstract DynamicJsonValue GetCounterRaw(string docId, string name);
 
-        protected (Document Document, List<Document> List) GetProjection(ref RetrieverInput retrieverInput, string lowerId, CancellationToken token)
+        protected (Document Document, List<Document> List) GetProjection(ref RetrieverInput retrieverInput, FieldsToFetch fieldsToFetch, string lowerId, CancellationToken token)
```

Comment:
Why `fieldsToFetch` is passed as parameter now? I see that in all usages we pass `FieldsToFetch` field.

Reply:
To unify and extract those methods to be able to be used for highlighting as a special codepath.